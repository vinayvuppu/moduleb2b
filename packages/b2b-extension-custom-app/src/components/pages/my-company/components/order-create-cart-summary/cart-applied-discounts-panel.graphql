query FetchDiscountCodesQuery($where: String!) {
  discountCodes(where: $where) {
    results {
      id
      code
      nameAllLocales {
        locale
        value
      }
      cartDiscounts {
        id
      }
    }
  }
}
