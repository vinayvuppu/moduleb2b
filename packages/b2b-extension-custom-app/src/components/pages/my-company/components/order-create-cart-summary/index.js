export { default } from './order-create-cart-summary';
