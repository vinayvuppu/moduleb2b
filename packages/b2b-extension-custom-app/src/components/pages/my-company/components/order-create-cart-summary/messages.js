import { defineMessages } from 'react-intl';

export default defineMessages({
  itemsLabel: {
    id: 'CartSummary.itemsLabel',
    description: 'items label when items are selected',
    defaultMessage: '{items} items',
  },
  cartSummaryTitle: {
    id: 'CartSummary.cartSummaryTitle',
    description: 'title for the cart summary panel',
    defaultMessage: 'in your cart',
  },
  quantityLabel: {
    id: 'CartSummary.quantityLabel',
    description: 'label for the line item quantity',
    defaultMessage: 'Qty: {quantity}',
  },
  shippingLabel: {
    id: 'CartSummary.shippingLabel',
    description: 'label for the shipping',
    defaultMessage: 'Shipping(gross)',
  },
  subtotalLabel: {
    id: 'CartSummary.subtotalLabel',
    description: 'label for the cart subtotal',
    defaultMessage: 'Subtotal',
  },
  subtotalDiscountsLabel: {
    id: 'CartSummary.subtotalDiscountsLabel',
    description: 'label for the cart subtotal after discounts applied',
    defaultMessage: 'Subtotal Discounts',
  },
  totalLabel: {
    id: 'CartSummary.totalLabel',
    description: 'label for the cart total',
    defaultMessage: 'Total (gross)',
  },
  originalTotalPriceLabel: {
    id: 'CartSummary.originalTotalPrice',
    description: 'label for the cart total original price',
    defaultMessage: 'Total original price',
  },
  emptyShoppingCartLabel: {
    id: 'CartSummary.emptyShoppingCartLabel',
    description: 'label for the empty shopping cart message',
    defaultMessage: 'Your shopping cart is empty',
  },
  removeDiscount: {
    id: 'CartSummary.removeDiscount',
    defaultMessage: 'Remove discount code',
  },
  discountCodes: {
    id: 'CartSummary.discountCodes',
    defaultMessage: 'Discount codes',
  },
  removeLineItem: {
    id: 'CartSummary.removeLineItem',
    defaultMessage: 'Remove Line item',
  },
  changeQuantity: {
    id: 'CartSummary.changeQuantity',
    defaultMessage: 'Change quantity',
  },
  changeQuantityTitle: {
    id: 'CartSummary.changeQuantityTitle',
    defaultMessage: 'Change {name} quantity',
  },
});
