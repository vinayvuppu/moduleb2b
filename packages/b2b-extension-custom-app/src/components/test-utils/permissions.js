import { PERMISSIONS } from '../../constants';

// eslint-disable-next-line import/prefer-default-export
export const allTestPermissions = Object.keys(PERMISSIONS).reduce(
  (permissionsMap, permissionKey) => ({
    ...permissionsMap,
    [`can${permissionKey}`]: true,
  }),
  {}
);
