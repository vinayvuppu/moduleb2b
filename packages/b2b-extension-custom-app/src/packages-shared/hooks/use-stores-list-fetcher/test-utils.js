import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import { FetchStoresQuery } from './stores-list-query.graphql';

// eslint-disable-next-line import/prefer-default-export
export const createFetchStoresListQueryMock = (custom = {}) => ({
  request: {
    variables: {
      target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
      limit: 20,
      offset: 0,
      sort: `createdAt desc`,
    },
    ...custom.request,
    query: FetchStoresQuery,
  },
  result: {
    data: {
      stores: {
        count: 2,
        offset: 0,
        total: 2,
        results: [
          {
            id: 'store-1',
            key: 'germany',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'Germany' }],
            createdAt: '2018-12-19T15:26:51.174Z',
            lastModifiedAt: '2019-03-01T09:00:51.174Z',
          },
          {
            id: 'store-2',
            key: 'usa',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'USA' }],
            createdAt: '2019-06-10T15:00:51.140Z',
            lastModifiedAt: '2019-08-30T10:26:51.140Z',
          },
        ],
      },
    },
    ...custom.result,
  },
});
