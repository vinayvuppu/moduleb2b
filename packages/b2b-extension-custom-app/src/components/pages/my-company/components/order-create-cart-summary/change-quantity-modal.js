import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { Spacings } from '@commercetools-frontend/ui-kit';

import { FormDialog } from '@commercetools-frontend/application-components';
import QuantitySelector from '../quantity-selector';
import messages from './messages';

const propTypes = {
  quantity: PropTypes.number,
  isOpen: PropTypes.bool.isRequired,
  productName: PropTypes.string,
  handleOnClose: PropTypes.func.isRequired,
  handleOnSave: PropTypes.func.isRequired,
};

const ChangeQuantityModal = props => {
  const intl = useIntl();

  const [quantity, setQuantity] = useState(props.quantity.toString());
  return (
    <FormDialog
      title={intl.formatMessage(messages.changeQuantityTitle, {
        name: props.productName,
      })}
      isOpen={props.isOpen}
      onClose={props.handleOnClose}
      onSecondaryButtonClick={props.handleOnClose}
      onPrimaryButtonClick={() => props.handleOnSave(quantity)}
      isPrimaryButtonDisabled={false}
    >
      <Spacings.Inline scale="s">
        <QuantitySelector
          onChange={setQuantity}
          quantity={quantity}
          maxValue={10}
          menuPortalTarget={document.body}
          menuPortalZIndex={1000}
        />
      </Spacings.Inline>
    </FormDialog>
  );
};

ChangeQuantityModal.displayName = 'ChangeQuantityModal';
ChangeQuantityModal.propTypes = propTypes;

export default ChangeQuantityModal;
