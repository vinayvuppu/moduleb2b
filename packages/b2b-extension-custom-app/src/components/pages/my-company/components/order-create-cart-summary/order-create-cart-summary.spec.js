import React from 'react';
import { shallow } from 'enzyme';
import { intlMock } from '@commercetools-local/test-utils';
import { OrderCreateCartSummary } from './order-create-cart-summary';

const createTestProps = props => ({
  cartDraft: {
    shippingInfo: undefined,
    totalPrice: {
      fractionDigits: 2,
    },
  },
  allLineItems: [],
  discounts: [],
  totalGross: {
    centAmount: 0,
    currencyCode: 'EUR',
  },
  subtotal: {
    centAmount: 0,
    currencyCode: 'EUR',
  },
  taxPortions: [],
  onRemoveDiscountCode: jest.fn(),
  onRemoveLineItem: jest.fn(),
  onChangeLineItemQuantity: jest.fn(),
  intl: intlMock,
  // withApplicationContext
  language: 'en',
  languages: ['en'],
  dataLocale: 'en',
  discountCodesQuery: {
    loading: false,
    discountCodes: {
      results: [],
    },
  },
  ...props,
});

describe('render', () => {
  describe('when the cart is empty', () => {
    let props;
    let wrapper;
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<OrderCreateCartSummary {...props} />);
    });
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when the cart is not empty', () => {
    describe('when no cart discounts are set to the cart', () => {
      let props;
      let wrapper;
      beforeEach(() => {
        props = createTestProps({
          cartDraft: {
            shippingInfo: {
              shippingMethodName: 'DHL',
            },
            totalPrice: {
              fractionDigits: 2,
            },
          },
          shippingPrice: {
            centAmount: 100,
            currencyCode: 'EUR',
          },
          totalGross: {
            centAmount: 2000,
            currencyCode: 'EUR',
          },
          subtotal: {
            centAmount: 1800,
            currencyCode: 'EUR',
          },
          taxPortions: [
            {
              name: 'tax portion',
              amount: {
                centAmount: 200,
                currencyCode: 'EUR',
              },
              rate: 0.12,
            },
          ],
          allLineItems: [
            {
              id: 'line-item-id',
              name: {
                en: 'item name',
              },
              quantity: 1,
              price: {
                value: {
                  centAmount: 1800,
                  currencyCode: 'EUR',
                },
              },
              totalPrice: {
                centAmount: 1800,
                currencyCode: 'EUR',
              },
              discountedPricePerQuantity: [],
              variant: { attributesRaw: [] },
            },
          ],
        });
        wrapper = shallow(<OrderCreateCartSummary {...props} />);
      });
      it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
    describe('when cart discounts are set to the cart', () => {
      let props;
      let wrapper;
      beforeEach(() => {
        props = createTestProps({
          cartDraft: {
            shippingInfo: {
              shippingMethodName: 'DHL',
            },
            totalPrice: {
              fractionDigits: 2,
            },
          },
          shippingPrice: {
            centAmount: 100,
            currencyCode: 'EUR',
          },
          totalGross: {
            centAmount: 2000,
            currencyCode: 'EUR',
          },
          subtotal: {
            centAmount: 1800,
            currencyCode: 'EUR',
          },
          taxPortions: [
            {
              name: 'tax portion',
              amount: {
                centAmount: 200,
                currencyCode: 'EUR',
              },
              rate: 0.12,
            },
          ],
          allLineItems: [
            {
              id: 'line-item-id',
              name: {
                en: 'item name',
              },
              quantity: 1,
              price: {
                value: {
                  centAmount: 1800,
                  currencyCode: 'EUR',
                },
              },
              totalPrice: {
                centAmount: 1800,
                currencyCode: 'EUR',
              },
              discountedPricePerQuantity: [],
              variant: { attributesRaw: [] },
            },
          ],
          discounts: [
            {
              id: 'cart-discount-id',
              amount: {
                centAmount: 500,
                currencyCode: 'EUR',
              },
              name: { en: 'cart-discount-name' },
            },
          ],
          discountCodesQuery: {
            loading: false,
            discountCodes: {
              results: [{ id: 'discount-code-id', code: 'code-1' }],
            },
          },
        });
        wrapper = shallow(<OrderCreateCartSummary {...props} />);
      });
      it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('when the change quantity modal is open', () => {
      let props;
      let wrapper;
      beforeEach(() => {
        props = createTestProps({
          cartDraft: {
            shippingInfo: {
              shippingMethodName: 'DHL',
            },
            totalPrice: {
              fractionDigits: 2,
            },
          },
          shippingPrice: {
            centAmount: 100,
            currencyCode: 'EUR',
          },
          totalGross: {
            centAmount: 2000,
            currencyCode: 'EUR',
          },
          subtotal: {
            centAmount: 1800,
            currencyCode: 'EUR',
          },
          taxPortions: [
            {
              name: 'tax portion',
              amount: {
                centAmount: 200,
                currencyCode: 'EUR',
              },
              rate: 0.12,
            },
          ],
          allLineItems: [
            {
              id: 'line-item-id',
              name: {
                en: 'item name',
              },
              quantity: 1,
              price: {
                value: {
                  centAmount: 1800,
                  currencyCode: 'EUR',
                },
              },
              totalPrice: {
                centAmount: 1800,
                currencyCode: 'EUR',
              },
              discountedPricePerQuantity: [],
              variant: { attributesRaw: [] },
            },
          ],
        });
        wrapper = shallow(<OrderCreateCartSummary {...props} />);
        wrapper.setState({
          isChangeQuantityModalOpen: true,
          changeQuantityLineItem: {
            id: 'id-1',
            quantity: 1,
            name: 'product-name',
          },
        });
      });
      it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});

describe('callbacks', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = createTestProps({
      cartDraft: {
        shippingInfo: {
          shippingMethodName: 'DHL',
        },
        totalPrice: {
          fractionDigits: 2,
        },
      },
      shippingPrice: {
        centAmount: 100,
        currencyCode: 'EUR',
      },
      totalGross: {
        centAmount: 2000,
        currencyCode: 'EUR',
      },
      subtotal: {
        centAmount: 1800,
        currencyCode: 'EUR',
      },
      taxPortions: [
        {
          name: 'tax portion',
          amount: {
            centAmount: 200,
            currencyCode: 'EUR',
          },
          rate: 0.12,
        },
      ],
      allLineItems: [
        {
          id: 'line-item-id',
          name: {
            en: 'item name',
          },
          quantity: 1,
          price: {
            value: {
              centAmount: 1800,
              currencyCode: 'EUR',
            },
          },
          totalPrice: {
            centAmount: 1800,
            currencyCode: 'EUR',
          },
          discountedPricePerQuantity: [],
          variant: { attributesRaw: [] },
        },
      ],
      discounts: [
        {
          id: 'cart-discount-id',
          amount: {
            centAmount: 500,
            currencyCode: 'EUR',
          },
          name: { en: 'cart-discount-name' },
        },
      ],
      discountCodesQuery: {
        loading: false,
        discountCodes: {
          results: [{ id: 'discount-code-id', code: 'code-1' }],
        },
      },
    });
    wrapper = shallow(<OrderCreateCartSummary {...props} />);
  });
  describe('when remove line item', () => {
    beforeEach(() => {
      wrapper
        .find({ 'data-testid': 'remove-line-item-line-item-id' })
        .simulate('click');
    });

    it('should call to onRemoveLineItem', () => {
      expect(props.onRemoveLineItem).toHaveBeenCalledWith('line-item-id');
    });
  });
  describe('when remove discount', () => {
    beforeEach(() => {
      wrapper
        .find({ 'data-testid': 'remove-discount-discount-code-id' })
        .simulate('click');
    });

    it('should call to onRemoveDiscountCode', () => {
      expect(props.onRemoveDiscountCode).toHaveBeenCalledWith(
        'discount-code-id'
      );
    });
  });

  describe('handleSelectLineItemToChangeQuantity', () => {
    beforeEach(() => {
      wrapper
        .instance()
        .handleSelectLineItemToChangeQuantity({ id: 'lineItemId' });
    });

    it('should change the isChangeQuantityModalOpen state', () => {
      expect(wrapper.state('isChangeQuantityModalOpen')).toEqual(true);
    });

    it('should change the changeQuantityLineItem state', () => {
      expect(wrapper.state('changeQuantityLineItem')).toEqual({
        id: 'lineItemId',
      });
    });
  });
  describe('handleCloseModal', () => {
    beforeEach(() => {
      wrapper.instance().handleCloseModal();
    });

    it('should change the isChangeQuantityModalOpen state', () => {
      expect(wrapper.state('isChangeQuantityModalOpen')).toEqual(false);
    });

    it('should change the changeQuantityLineItem state', () => {
      expect(wrapper.state('changeQuantityLineItem')).toEqual(undefined);
    });
  });

  describe('handleChangeQuantity', () => {
    let instance;
    beforeEach(() => {
      instance = wrapper.instance();
      instance.handleSelectLineItemToChangeQuantity({
        id: 'lineItemId',
        quantity: 10,
      });

      instance.handleChangeQuantity(29);
    });

    it('should call to onChangeLineItemQuantity', () => {
      expect(props.onChangeLineItemQuantity).toHaveBeenCalledWith(
        'lineItemId',
        29
      );
    });

    it('should change the isChangeQuantityModalOpen state', () => {
      expect(wrapper.state('isChangeQuantityModalOpen')).toEqual(false);
    });

    it('should change the changeQuantityLineItem state', () => {
      expect(wrapper.state('changeQuantityLineItem')).toEqual(undefined);
    });
  });
});
