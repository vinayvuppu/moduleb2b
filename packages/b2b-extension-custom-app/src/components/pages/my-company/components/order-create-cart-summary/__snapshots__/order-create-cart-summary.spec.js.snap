// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when the cart is empty should match snapshot 1`] = `
<Card
  theme="light"
  type="raised"
>
  <Stack
    alignItems="stretch"
    scale="s"
  >
    <Inline
      alignItems="center"
      justifyContent="flex-start"
      scale="xs"
    >
      <CartIcon
        size="medium"
      />
      <TextDetail
        intlMessage={
          Object {
            "defaultMessage": "{items} items",
            "description": "items label when items are selected",
            "id": "CartSummary.itemsLabel",
            "values": Object {
              "items": 0,
            },
          }
        }
        isBold={true}
        tone="information"
      />
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="in your cart"
          description="title for the cart summary panel"
          id="CartSummary.cartSummaryTitle"
          values={Object {}}
        />
      </TextDetail>
    </Inline>
    <div
      className="separator"
    />
    <TextDetail
      tone="secondary"
    >
      <FormattedMessage
        defaultMessage="Your shopping cart is empty"
        description="label for the empty shopping cart message"
        id="CartSummary.emptyShoppingCartLabel"
        values={Object {}}
      />
    </TextDetail>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Subtotal"
          description="label for the cart subtotal"
          id="CartSummary.subtotalLabel"
          values={Object {}}
        />
      </TextDetail>
      <TextDetail
        isBold={true}
      >
        EUR 0
      </TextDetail>
    </SummaryItem>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextBody
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Total (gross)"
          description="label for the cart total"
          id="CartSummary.totalLabel"
          values={Object {}}
        />
      </TextBody>
      <TextBody
        isBold={true}
      >
        EUR 0
      </TextBody>
    </SummaryItem>
  </Stack>
</Card>
`;

exports[`render when the cart is not empty when cart discounts are set to the cart should match snapshot 1`] = `
<Card
  theme="light"
  type="raised"
>
  <Stack
    alignItems="stretch"
    scale="s"
  >
    <Inline
      alignItems="center"
      justifyContent="flex-start"
      scale="xs"
    >
      <CartIcon
        size="medium"
      />
      <TextDetail
        intlMessage={
          Object {
            "defaultMessage": "{items} items",
            "description": "items label when items are selected",
            "id": "CartSummary.itemsLabel",
            "values": Object {
              "items": 1,
            },
          }
        }
        isBold={true}
        tone="information"
      />
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="in your cart"
          description="title for the cart summary panel"
          id="CartSummary.cartSummaryTitle"
          values={Object {}}
        />
      </TextDetail>
    </Inline>
    <div
      className="separator"
    />
    <SummaryItem
      key="line-item-id"
    >
      <Inline
        alignItems="flex-start"
        justifyContent="flex-start"
        scale="s"
      >
        <IconButton
          data-testid="remove-line-item-line-item-id"
          icon={<CloseBoldIcon />}
          isToggleButton={false}
          label="CartSummary.removeLineItem"
          onClick={[Function]}
          shape="round"
          size="small"
          theme="default"
          type="button"
        />
        <Stack
          alignItems="stretch"
          scale="s"
        >
          <TextDetail
            isBold={true}
          >
            <withApplicationContext(Localize)
              object={
                Object {
                  "discountedPricePerQuantity": Array [],
                  "id": "line-item-id",
                  "name": Object {
                    "en": "item name",
                  },
                  "price": Object {
                    "value": Object {
                      "centAmount": 1800,
                      "currencyCode": "EUR",
                    },
                  },
                  "quantity": 1,
                  "totalPrice": Object {
                    "centAmount": 1800,
                    "currencyCode": "EUR",
                  },
                  "variant": Object {
                    "attributesRaw": Array [],
                  },
                }
              }
              objectKey="name"
            />
          </TextDetail>
        </Stack>
      </Inline>
      <Stack
        alignItems="flexEnd"
        scale="s"
      >
        <TextDetail
          tone="secondary"
        >
          EUR 18
        </TextDetail>
        <Inline
          alignItems="flex-start"
          justifyContent="flex-start"
          scale="s"
        >
          <IconButton
            icon={<EditIcon />}
            isToggleButton={false}
            label="CartSummary.changeQuantity"
            onClick={[Function]}
            shape="round"
            size="small"
            theme="default"
            type="button"
          />
          <TextDetail
            isBold={true}
          >
            <FormattedMessage
              defaultMessage="Qty: {quantity}"
              description="label for the line item quantity"
              id="CartSummary.quantityLabel"
              values={
                Object {
                  "quantity": 1,
                }
              }
            />
          </TextDetail>
        </Inline>
        <TextDetail
          isBold={true}
        >
          EUR 18
        </TextDetail>
      </Stack>
    </SummaryItem>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Subtotal"
          description="label for the cart subtotal"
          id="CartSummary.subtotalLabel"
          values={Object {}}
        />
      </TextDetail>
      <TextDetail
        isBold={true}
      >
        EUR 23
      </TextDetail>
    </SummaryItem>
    <SummaryItem
      key="cart-discount-id"
    >
      <TextDetail
        tone="secondary"
      >
        <withApplicationContext(Localize)
          object={
            Object {
              "amount": Object {
                "centAmount": 500,
                "currencyCode": "EUR",
              },
              "id": "cart-discount-id",
              "name": Object {
                "en": "cart-discount-name",
              },
            }
          }
          objectKey="name"
        />
      </TextDetail>
      <TextDetail
        tone="secondary"
      >
        - EUR 5
      </TextDetail>
    </SummaryItem>
    <Stack
      alignItems="stretch"
      scale="s"
    >
      <div
        className="separator"
      />
      <SummaryItem>
        <TextDetail
          isBold={true}
        >
          <FormattedMessage
            defaultMessage="Discount codes"
            id="CartSummary.discountCodes"
            values={Object {}}
          />
        </TextDetail>
      </SummaryItem>
      <SummaryItem
        key="discount-code-id"
      >
        <Inline
          alignItems="flex-start"
          justifyContent="flex-start"
          scale="s"
        >
          <IconButton
            data-testid="remove-discount-discount-code-id"
            icon={<CloseBoldIcon />}
            isToggleButton={false}
            label="CartSummary.removeDiscount"
            onClick={[Function]}
            shape="round"
            size="small"
            theme="default"
            type="button"
          />
          <TextDetail
            tone="secondary"
          >
            code-1
          </TextDetail>
        </Inline>
      </SummaryItem>
    </Stack>
    <Stack
      alignItems="stretch"
      scale="s"
    >
      <div
        className="separator"
      />
      <SummaryItem>
        <TextDetail
          isBold={true}
        >
          <FormattedMessage
            defaultMessage="Subtotal Discounts"
            description="label for the cart subtotal after discounts applied"
            id="CartSummary.subtotalDiscountsLabel"
            values={Object {}}
          />
        </TextDetail>
        <TextDetail
          isBold={true}
        >
          EUR 18
        </TextDetail>
      </SummaryItem>
    </Stack>
    <SummaryItem
      key="0"
    >
      <TextDetail
        tone="secondary"
      >
        12% tax portion
      </TextDetail>
      <TextDetail
        tone="secondary"
      >
        + EUR 2
      </TextDetail>
    </SummaryItem>
    <SummaryItem>
      <Stack
        alignItems="stretch"
        scale="xs"
      >
        <TextDetail
          tone="secondary"
        >
          <FormattedMessage
            defaultMessage="Shipping(gross)"
            description="label for the shipping"
            id="CartSummary.shippingLabel"
            values={Object {}}
          />
        </TextDetail>
        <TextDetail
          tone="secondary"
        >
          DHL
        </TextDetail>
      </Stack>
      <TextDetail
        tone="secondary"
      >
        + EUR 1
      </TextDetail>
    </SummaryItem>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextBody
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Total (gross)"
          description="label for the cart total"
          id="CartSummary.totalLabel"
          values={Object {}}
        />
      </TextBody>
      <TextBody
        isBold={true}
      >
        EUR 20
      </TextBody>
    </SummaryItem>
  </Stack>
</Card>
`;

exports[`render when the cart is not empty when no cart discounts are set to the cart should match snapshot 1`] = `
<Card
  theme="light"
  type="raised"
>
  <Stack
    alignItems="stretch"
    scale="s"
  >
    <Inline
      alignItems="center"
      justifyContent="flex-start"
      scale="xs"
    >
      <CartIcon
        size="medium"
      />
      <TextDetail
        intlMessage={
          Object {
            "defaultMessage": "{items} items",
            "description": "items label when items are selected",
            "id": "CartSummary.itemsLabel",
            "values": Object {
              "items": 1,
            },
          }
        }
        isBold={true}
        tone="information"
      />
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="in your cart"
          description="title for the cart summary panel"
          id="CartSummary.cartSummaryTitle"
          values={Object {}}
        />
      </TextDetail>
    </Inline>
    <div
      className="separator"
    />
    <SummaryItem
      key="line-item-id"
    >
      <Inline
        alignItems="flex-start"
        justifyContent="flex-start"
        scale="s"
      >
        <IconButton
          data-testid="remove-line-item-line-item-id"
          icon={<CloseBoldIcon />}
          isToggleButton={false}
          label="CartSummary.removeLineItem"
          onClick={[Function]}
          shape="round"
          size="small"
          theme="default"
          type="button"
        />
        <Stack
          alignItems="stretch"
          scale="s"
        >
          <TextDetail
            isBold={true}
          >
            <withApplicationContext(Localize)
              object={
                Object {
                  "discountedPricePerQuantity": Array [],
                  "id": "line-item-id",
                  "name": Object {
                    "en": "item name",
                  },
                  "price": Object {
                    "value": Object {
                      "centAmount": 1800,
                      "currencyCode": "EUR",
                    },
                  },
                  "quantity": 1,
                  "totalPrice": Object {
                    "centAmount": 1800,
                    "currencyCode": "EUR",
                  },
                  "variant": Object {
                    "attributesRaw": Array [],
                  },
                }
              }
              objectKey="name"
            />
          </TextDetail>
        </Stack>
      </Inline>
      <Stack
        alignItems="flexEnd"
        scale="s"
      >
        <TextDetail
          tone="secondary"
        >
          EUR 18
        </TextDetail>
        <Inline
          alignItems="flex-start"
          justifyContent="flex-start"
          scale="s"
        >
          <IconButton
            icon={<EditIcon />}
            isToggleButton={false}
            label="CartSummary.changeQuantity"
            onClick={[Function]}
            shape="round"
            size="small"
            theme="default"
            type="button"
          />
          <TextDetail
            isBold={true}
          >
            <FormattedMessage
              defaultMessage="Qty: {quantity}"
              description="label for the line item quantity"
              id="CartSummary.quantityLabel"
              values={
                Object {
                  "quantity": 1,
                }
              }
            />
          </TextDetail>
        </Inline>
        <TextDetail
          isBold={true}
        >
          EUR 18
        </TextDetail>
      </Stack>
    </SummaryItem>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Subtotal"
          description="label for the cart subtotal"
          id="CartSummary.subtotalLabel"
          values={Object {}}
        />
      </TextDetail>
      <TextDetail
        isBold={true}
      >
        EUR 18
      </TextDetail>
    </SummaryItem>
    <SummaryItem
      key="0"
    >
      <TextDetail
        tone="secondary"
      >
        12% tax portion
      </TextDetail>
      <TextDetail
        tone="secondary"
      >
        + EUR 2
      </TextDetail>
    </SummaryItem>
    <SummaryItem>
      <Stack
        alignItems="stretch"
        scale="xs"
      >
        <TextDetail
          tone="secondary"
        >
          <FormattedMessage
            defaultMessage="Shipping(gross)"
            description="label for the shipping"
            id="CartSummary.shippingLabel"
            values={Object {}}
          />
        </TextDetail>
        <TextDetail
          tone="secondary"
        >
          DHL
        </TextDetail>
      </Stack>
      <TextDetail
        tone="secondary"
      >
        + EUR 1
      </TextDetail>
    </SummaryItem>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextBody
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Total (gross)"
          description="label for the cart total"
          id="CartSummary.totalLabel"
          values={Object {}}
        />
      </TextBody>
      <TextBody
        isBold={true}
      >
        EUR 20
      </TextBody>
    </SummaryItem>
  </Stack>
</Card>
`;

exports[`render when the cart is not empty when the change quantity modal is open should match snapshot 1`] = `
<Card
  theme="light"
  type="raised"
>
  <Stack
    alignItems="stretch"
    scale="s"
  >
    <Inline
      alignItems="center"
      justifyContent="flex-start"
      scale="xs"
    >
      <CartIcon
        size="medium"
      />
      <TextDetail
        intlMessage={
          Object {
            "defaultMessage": "{items} items",
            "description": "items label when items are selected",
            "id": "CartSummary.itemsLabel",
            "values": Object {
              "items": 1,
            },
          }
        }
        isBold={true}
        tone="information"
      />
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="in your cart"
          description="title for the cart summary panel"
          id="CartSummary.cartSummaryTitle"
          values={Object {}}
        />
      </TextDetail>
    </Inline>
    <div
      className="separator"
    />
    <SummaryItem
      key="line-item-id"
    >
      <Inline
        alignItems="flex-start"
        justifyContent="flex-start"
        scale="s"
      >
        <IconButton
          data-testid="remove-line-item-line-item-id"
          icon={<CloseBoldIcon />}
          isToggleButton={false}
          label="CartSummary.removeLineItem"
          onClick={[Function]}
          shape="round"
          size="small"
          theme="default"
          type="button"
        />
        <Stack
          alignItems="stretch"
          scale="s"
        >
          <TextDetail
            isBold={true}
          >
            <withApplicationContext(Localize)
              object={
                Object {
                  "discountedPricePerQuantity": Array [],
                  "id": "line-item-id",
                  "name": Object {
                    "en": "item name",
                  },
                  "price": Object {
                    "value": Object {
                      "centAmount": 1800,
                      "currencyCode": "EUR",
                    },
                  },
                  "quantity": 1,
                  "totalPrice": Object {
                    "centAmount": 1800,
                    "currencyCode": "EUR",
                  },
                  "variant": Object {
                    "attributesRaw": Array [],
                  },
                }
              }
              objectKey="name"
            />
          </TextDetail>
        </Stack>
      </Inline>
      <Stack
        alignItems="flexEnd"
        scale="s"
      >
        <TextDetail
          tone="secondary"
        >
          EUR 18
        </TextDetail>
        <Inline
          alignItems="flex-start"
          justifyContent="flex-start"
          scale="s"
        >
          <IconButton
            icon={<EditIcon />}
            isToggleButton={false}
            label="CartSummary.changeQuantity"
            onClick={[Function]}
            shape="round"
            size="small"
            theme="default"
            type="button"
          />
          <TextDetail
            isBold={true}
          >
            <FormattedMessage
              defaultMessage="Qty: {quantity}"
              description="label for the line item quantity"
              id="CartSummary.quantityLabel"
              values={
                Object {
                  "quantity": 1,
                }
              }
            />
          </TextDetail>
        </Inline>
        <TextDetail
          isBold={true}
        >
          EUR 18
        </TextDetail>
      </Stack>
    </SummaryItem>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextDetail
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Subtotal"
          description="label for the cart subtotal"
          id="CartSummary.subtotalLabel"
          values={Object {}}
        />
      </TextDetail>
      <TextDetail
        isBold={true}
      >
        EUR 18
      </TextDetail>
    </SummaryItem>
    <SummaryItem
      key="0"
    >
      <TextDetail
        tone="secondary"
      >
        12% tax portion
      </TextDetail>
      <TextDetail
        tone="secondary"
      >
        + EUR 2
      </TextDetail>
    </SummaryItem>
    <SummaryItem>
      <Stack
        alignItems="stretch"
        scale="xs"
      >
        <TextDetail
          tone="secondary"
        >
          <FormattedMessage
            defaultMessage="Shipping(gross)"
            description="label for the shipping"
            id="CartSummary.shippingLabel"
            values={Object {}}
          />
        </TextDetail>
        <TextDetail
          tone="secondary"
        >
          DHL
        </TextDetail>
      </Stack>
      <TextDetail
        tone="secondary"
      >
        + EUR 1
      </TextDetail>
    </SummaryItem>
    <div
      className="separator"
    />
    <SummaryItem>
      <TextBody
        isBold={true}
      >
        <FormattedMessage
          defaultMessage="Total (gross)"
          description="label for the cart total"
          id="CartSummary.totalLabel"
          values={Object {}}
        />
      </TextBody>
      <TextBody
        isBold={true}
      >
        EUR 20
      </TextBody>
    </SummaryItem>
  </Stack>
  <ChangeQuantityModal
    handleOnClose={[Function]}
    handleOnSave={[Function]}
    isOpen={true}
    productName="product-name"
    quantity={1}
  />
</Card>
`;
